src/BatteryStateOfChargeService.cpp

```diff
@@ -16,21 +24,57 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return BATTERY_AMP_HOUR_CAPACITY-ampHours ;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+    if(previousCurrent<0)
+    {
     return false;
+    }
+    else
+    {
+       return true;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+
+    double timeRemaining = (BATTERY_AMP_HOUR_CAPACITY-ampHours)/previousCurrent ;
+
+    timeRemaining = qAbs(timeRemaining)/MILISECONDS_TO_HOURS;
+
+    QTime time(0, 0, 0, 0);
+
+    QTime t;
+
+    t = time.addMSecs(timeRemaining);
+
+    return t;
```

Comment:
This variable t is not really necessary, lines 49 to 55 could be compressed into one line:
```c++
return QTime(0, 0).addMSecs(timeRemaining);
```